utils/Azure/acrTools.ts

```diff
@@ -0,0 +1,277 @@
+import { Registry } from "azure-arm-containerregistry/lib/models";
+import { SubscriptionModels } from 'azure-arm-resource';
+import request = require('request-promise');
+import * as vscode from "vscode";
+import { AzureImageNode, AzureRepositoryNode } from '../../explorer/models/AzureRegistryNodes';
+import { AzureAccount, AzureSession } from "../../typings/azure-account.api";
+import { AzureImage } from "../Azure/models/image";
+import { Repository } from "../Azure/models/Repository";
+import { AzureCredentialsManager } from '../azureCredentialsManager';
+const teleCmdId: string = 'vscode-docker.deleteAzureImage';
+
+/**
+ * Developers can use this to visualize and list repositories on a given Registry. This is not a command, just a developer tool.
+ * @param registry : the registry whose repositories you want to see
+ * @returns allRepos : an array of Repository objects that exist within the given registry
+ */
+export async function getAzureRepositories(registry: Registry): Promise<Repository[]> {
+    const allRepos: Repository[] = [];
+    let repo: Repository;
+    let azureAccount: AzureAccount = AzureCredentialsManager.getInstance().getAccount();
+    if (!azureAccount) {
+        return [];
+    }
+    const { accessToken, refreshToken } = await getTokens(registry);
+    if (accessToken && refreshToken) {
+
+        await request.get('https://' + registry.loginServer + '/v2/_catalog', {
+            auth: {
+                bearer: accessToken
+            }
```

Comment:
Can it use request_data_from_registry?

Reply:
no, request data from registry actually deletes something, and we don't want to delete anything here